dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeMetadataConnector.java

```diff
@@ -122,18 +124,23 @@ protected void addSqlTasks(
       @Nonnull Class<? extends Enum<?>> header,
       @Nonnull String format,
       @Nonnull TaskVariant is_task,
-      @Nonnull TaskVariant au_task) {
-    Task<?> t0 =
+      @Nonnull TaskVariant au_task,
+      ConnectorArguments arguments) {
+    AbstractJdbcTask<Summary> is_jdbcTask =
         new JdbcSelectTask(
                 is_task.zipEntryName,
                 String.format(format, is_task.schemaName, is_task.whereClause))
             .withHeaderClass(header);
-    Task<?> t1 =
+
+    AbstractJdbcTask<Summary> au_jdbcTask =
         new JdbcSelectTask(
                 au_task.zipEntryName,
                 String.format(format, au_task.schemaName, au_task.whereClause))
-            .withHeaderClass(header)
-            .onlyIfFailed(t0);
+            .withHeaderClass(header);
+
+    Task<?> t0 = arguments.isAssessment() ? au_jdbcTask : is_jdbcTask;
+    Task<?> t1 = (arguments.isAssessment() ? is_jdbcTask : au_jdbcTask).onlyIfFailed(t0);
```

Comment:
I think that in the case of isAssessment=true, we do not even want to go with a fallback, because even if we can use the limited information from it, it creates a lot of complications for downstream processing.

Reply:
Understood. Modified accordingly